### ISSUE TITLE
Support code coverage / istanbul

### ISSUE THREAD

**Issue by Raynos** [testem/testem#229]
I've been using istanbul recently for server-side stuff and it's really nice.

Doing code coverage for client side requires a server to send client-side generate coverage reports to and then using server side tooling to convert coverage reports (json objects) into lcov or html. 

Having this functionality in testem would be really nice.

**Comment by jehoshua02** [comment]
+1 For :birthday: :gift:?

**Comment by laland** [comment]
+1.5

**Comment by elfet** [comment]
+2

**Comment by benoror** [comment]
+10

**Comment by cbuteau** [comment]
So I tried the home grown istanbul example project which currently does not work out of the box.  But it was a good starting point.

I figured out the actual commandline for the istanbul instrumentation luckliy my project is a single file.

```cmd
node ./node_modules/istanbul/lib/cli.js instrument <filename>.js --output instrument/<filename>.js
```

But when I tried to run testem in dev mode it would not work because it was launching the customized coverage testem.js config file.

I brought it up in node-debug and found I could exit before_tests and after_tests if 

```javascript

```
